aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -1467,7 +1468,8 @@ Tensor zeros(
     c10::optional<ScalarType> dtype,
     c10::optional<Layout> layout,
     c10::optional<Device> device,
-    c10::optional<bool> pin_memory) {
+    c10::optional<bool> pin_memory,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
```

Comment:
You added the extra argument, but it's not used anywhere?

Reply:
Since this version of `zeros` is implemented with a call to `full`, then you'd need to also add support for `full` if you want to add support for memory_format on this overload too.

I can see that `full` currently doesn't support memory_format by looking at its schema in the yaml here: https://github.com/pytorch/pytorch/blob/6a30d83596f0c0164d7f7431a10557239371281d/aten/src/ATen/native/native_functions.yaml#L2054

Reply:
It can be implemented like non-named overload above, with empty.names overload (that also accepts memory layout) and `zero_`